src/main/java/com/icthh/xm/ms/scheduler/manager/SchedulingManager.java

```diff
@@ -212,6 +218,25 @@ private boolean deleteTaskFromTenant(String tenant, String taskKey,
         return cancelled;
     }
 
+    /**
+     * Set state for task.
+     * Method init tenant context with value from input task because method called from another thread
+     *
+     * @param state new state of the task
+     * @param task  the task changing
+     */
+    public void setState(StateKey state, TaskDTO task) {
+        Long taskId = task.getId();
+        log.info("Marking task as finished id: {}, tenant: {}", task.getId(), task.getTenant());
+        TenantContextUtils.setTenant(tenantContextHolder, task.getTenant());
+
+        Task taskFromDb = taskRepository.findById(taskId).orElseThrow(()
+            -> new IllegalArgumentException("Entity is not found"));
```

Comment:
I propose to use BusinessException as produces http 400 error instead of 500 and provides message localization.

Reply:
ok, i will create another pull request

Reply:
https://github.com/xm-online/xm-ms-scheduler/pull/11